### ISSUE TITLE
Use %appdata% for stdout/stderr output

### ISSUE THREAD

**Issue by Zabeus** [nuvie/nuvie#53]
This is related to ticket #7. In many cases on modern Windows platforms the stderr.txt and stdout.txt files are not written anywhere obvious. They are just written to the program run directory by default. We can try redirecting SDL's output per

**Comment by DominusExult** [comment]
Can't we just link to the files in the VirtualStore folder where they are redirected to?
IMO for the few cases when normal users actually need the std files, it's a bit over the top to provide a special sdl.dll.

**Comment by DominusExult** [comment]
Furthermore, AFAI can tell, SDL2 doesn't produce those files anymore and the code for saving those should be in Nuvie

**Comment by Zabeus** [comment]
I've added code to my local branch to redirect output without recompiling SDL, and like wjp suspected it was pretty easy. Based on the above link I wonder if it will fail to work sometimes.
ALTHOUGH I thought SDL2 was already enabled by default. Didn't realize I was working with SDL1.

Due to malignant's experience I'm not 100% sure the VirtualStore is always in the same place. If he's using SDL2 maybe that's why they weren't being generated then.

**Comment by Zabeus** [comment]
I pushed the fix, but am unable to test with SDL2. @malignantmanor can you see if it generates stderr.txt now (with some contents)? Testing Branch at -> https://github.com/nuvie/nuvie/tree/stdoutput

It should go to some place like C:\Users\<your username>\AppData\Roaming\Nuvie\stderr.txt.

**Comment by malignantmanor** [comment]
That seems to work for SDL 2, but it doesn't print the same way. It stops printing at "Loading super chunk 62 of " and doesn't seem to print again until the buffer is full (or something) and then you have the same partial printing problem.

SDL 1.2 still writes the output to the exe directory and prints like normal. I did not test the installer.

**Comment by malignantmanor** [comment]
SDL 1.2 has the same printing issue. I'm not sure why you are having SDL 1.2 print out the redirection text every time the engine starts.

**Comment by Zabeus** [comment]
Thanks. You don't have to test the installer. If you mean "stdout/stderr redirected to ***", I made it print that because it was still generating the empty files in the exe directory and that way the user can find where the "real" files are. The message in the redirected file is more for debug/testing and should probably be removed.

**Comment by malignantmanor** [comment]
Is there a way to stop SDL from writing those files?

**Comment by Zabeus** [comment]
Yes there's some info about it in the page Dominus linked http://forums.libsdl.org/viewtopic.php?t=9454&sid=501bd8209741efc17132dbd5ef4444f7 and on stackoverflow but it involves recompiling SDLmain. Although hackish I'd rather delete the files or just leave them empty. I wonder if we need a flush() call added to fix the delay. Not sure why it works differently now.

**Comment by malignantmanor** [comment]
I'm pretty sure Exult prints them only in the app directory. They don't use SDLmain though.

**Comment by Zabeus** [comment]
Yep I found the code for this at https://github.com/rofl0r/exult/blob/master/files/utils.cc, line 633 redirect_output(). It uses system home path or local appdata. It also has a cleanup_output() function to delete them.
Also found a mention by you and Dominus of this in #exult logs from 2010.
